app/controllers/chargeback_controller.rb

```diff
@@ -296,8 +296,26 @@ def cb_rates_delete
           page.replace("flash_msg_div", :partial => "layouts/flash_msg")
         end
       end
-      process_cb_rates(rates, "destroy")  unless rates.empty?
-      add_flash(_("The selected %s were deleted") % ui_lookup(:models => "ChargebackRate"), :info, true) unless flash_errors?
+
+      # Do not delete assigned rates
+      rates_to_delete = []
+      rates.each do |cb_rate_id|
+        cb_rate = ChargebackRate.find_by_id(cb_rate_id)
+        if cb_rate.assigned?
+          rate_descr = cb_rate.description
+          add_flash(_("Selected %{model} %{rate} is assigned and cannot be deleted") % {:model => "ChargebackRate", :rate  => rate_descr}, :error)
+        else
+          rates_to_delete << cb_rate_id
+        end
+      end
+
+      process_cb_rates(rates_to_delete, "destroy")  unless rates_to_delete.empty?
+      if rates_to_delete.size > 1
+        add_flash(_("The selected %s were deleted") % ui_lookup(:models => "ChargebackRate"), :info, true) unless flash_errors?
+      else
+        add_flash(_("The selected %s was deleted") % ui_lookup(:model => "ChargebackRate"), :info, true) unless flash_errors?
+      end
```

Comment:
This seems like a lot for a plural check.  I think there's a nicer way to do this with I18N / ui_lookup.